nsx/resource_nsx_edge_dlr.go

```diff
@@ -0,0 +1,186 @@
+package nsx
+
+import (
+	"fmt"
+	"github.com/IBM-tfproviders/govnsx"
+	"github.com/IBM-tfproviders/govnsx/nsxresource"
+	"github.com/IBM-tfproviders/govnsx/nsxtypes"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+)
+
+const (
+	DLRResourceIdPrefix = "dlr-"
+)
+
+type ifCfg struct {
+	name              string
+	ip                string
+	mask              string
+	logical_switch_id string
+}
+
+type dlrCfg struct {
+	edgeId    string
+	ifCfgList []ifCfg
+}
+
+func resourceNsxEdgeDLR() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxEdgeDLRInterfaceCreate,
+		Read:   resourceNsxEdgeDLRInterfaceRead,
+		Update: resourceNsxEdgeDLRInterfaceUpdate,
+		Delete: resourceNsxEdgeDLRInterfaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"edge_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"interface": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				MaxItems: 999,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"index": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ip": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
```

Comment:
We might need to allow IP update for the interface instead of forcenew. For now Forenew is not bad.